test/e2e/aadpodidentity_test.go

```diff
@@ -626,6 +626,8 @@ func setUpIdentityAndDeployment(azureIdentityName, suffix, replicas string) {
 	ok, err = daemonset.WaitOnReady(nmiDaemonSet)
 	Expect(err).NotTo(HaveOccurred())
 	Expect(ok).To(Equal(true))
+
+	time.Sleep(30 * time.Second)
```

Comment:
we can remove this sleep in the states PR because there we can start waiting for n assigned identities with desired state.